graveler/committed/tree/common_parts_test.go

```diff
@@ -0,0 +1,105 @@
+)
+
+func TestRemoveCommonParts(t *testing.T) {
+	tests := []struct {
+		left                 *tree.Tree
+		right                *tree.Tree
+		expectedLeftPartIds  []string
+		expectedRightPartIds []string
+	}{
+		{
+			left:                 newTestTree([]string{"A", "B", "C"}, []string{"a9", "b9", "c9"}),
+			right:                newTestTree([]string{"A", "C"}, []string{"a9", "c9"}),
+			expectedLeftPartIds:  []string{"B"},
+			expectedRightPartIds: []string{},
+		},
+		{
+			left:                 newTestTree([]string{"A", "B", "C"}, []string{"a9", "b9", "c9"}),
+			right:                newTestTree([]string{"A", "B1", "C"}, []string{"a9", "b9", "c9"}),
+			expectedLeftPartIds:  []string{"B"},
+			expectedRightPartIds: []string{"B1"},
+		},
+		{
+			left:                 newTestTree([]string{"A", "B", "C"}, []string{"a9", "b99", "c9"}),
+			right:                newTestTree([]string{"A", "B1", "C"}, []string{"a9", "b9", "c9"}),
+			expectedLeftPartIds:  []string{"B"},
+			expectedRightPartIds: []string{"B1"},
+		},
+		{
+			left:                 newTestTree([]string{"A", "B", "C"}, []string{"a9", "b9", "c9"}),
+			right:                newTestTree([]string{"A", "B1", "C"}, []string{"a9", "b99", "c9"}),
+			expectedLeftPartIds:  []string{"B"},
+			expectedRightPartIds: []string{"B1"},
+		},
+		{
+			left:                 newTestTree([]string{"A", "B", "D"}, []string{"a9", "b9", "d9"}),
+			right:                newTestTree([]string{"A", "C", "D", "E"}, []string{"a9", "c9", "d9", "e9"}),
+			expectedLeftPartIds:  []string{"B"},
+			expectedRightPartIds: []string{"C", "E"},
+		},
+		{
+			left:                 newTestTree([]string{"A", "D", "E"}, []string{"a9", "d9", "e9"}),
+			right:                newTestTree([]string{"A", "B", "C", "D", "F"}, []string{"a9", "b9", "c9", "d9", "f9"}),
+			expectedLeftPartIds:  []string{"E"},
+			expectedRightPartIds: []string{"B", "C", "F"},
+		},
+		{
+			left:                 newTestTree([]string{"A", "B", "C", "D", "F"}, []string{"a9", "b9", "c9", "d9", "f9"}),
+			right:                newTestTree([]string{"A", "D", "E"}, []string{"a9", "d9", "e9"}),
+			expectedLeftPartIds:  []string{"B", "C", "F"},
+			expectedRightPartIds: []string{"E"},
+		},
+		{
+			left:                 newTestTree([]string{"Z"}, []string{"z9"}),
+			right:                newTestTree([]string{"A", "B", "C", "D", "E"}, []string{"a9", "b9", "c9", "d9", "e9"}),
+			expectedLeftPartIds:  []string{"Z"},
+			expectedRightPartIds: []string{"A", "B", "C", "D", "E"},
+		},
+		{
+			left:                 newTestTree([]string{"A", "B", "C", "D", "E"}, []string{"a9", "b9", "c9", "d9", "e9"}),
+			right:                newTestTree([]string{"Z"}, []string{"z9"}),
+			expectedLeftPartIds:  []string{"A", "B", "C", "D", "E"},
+			expectedRightPartIds: []string{"Z"},
+		},
+	}
+	for _, tst := range tests {
+		gotLeft, gotRight := tree.RemoveCommonParts(tst.left, tst.right)
+		if len(gotLeft.Parts) != len(tst.expectedLeftPartIds) {
```

Comment:
Think that if you use deep.Equal, one for left and one for the right it will be less code and easier output.